### PR TITLE
Whitehall Transition: Release 1.2

### PR DESCRIPTION
**Pull request by ollietreend**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

**Review by beccapearce — COMMENTED** [review]
Looking good to me!

File: features/new-tagging-workflow.feature [hunk 1]
```diff
@@ -18,24 +18,8 @@ Scenario: Publication that does not support the new taxonomy
   Then I should be on the legacy tagging page
   And I should be able to update the legacy tags
 
-Scenario: Publication that supports the new taxonomy
-  Given I am a writer
-  When I start editing a draft document which can be tagged to the new taxonomy
-  And I continue to the tagging page
-  Then I should be on the taxonomy tagging page
-  And I should be able to update the taxonomy and click the "Save tagging changes" button
```
Comment (DilwoarH) [inline features/new-tagging-workflow.feature:26]: Do we want to delete this?

File: lib/taxonomy/topic_taxonomy.rb [hunk 2]
```diff
@@ -37,11 +37,15 @@ def ordered_branches
...
+        checked = taxon.taxon_list.any? do |descendant_taxon|
+          selected_taxon_content_ids.include?(descendant_taxon.content_id)
+        end
+
         {
           label: taxon.name,
           value: taxon.content_id,
           items: (transform_taxon(taxon.children, selected_taxon_content_ids) if taxon.children),
-          checked: selected_taxon_content_ids.include?(taxon.content_id),
+          checked:,
```
Comment (DilwoarH) [inline lib/taxonomy/topic_taxonomy.rb:48]: is this correct?

Comment (DilwoarH) [reply lib/taxonomy/topic_taxonomy.rb:48]:
```suggestion
          checked: checked,
```

should it be this?

Comment (ollietreend) [reply lib/taxonomy/topic_taxonomy.rb:48]: Yes – so, initially this changed from:

```diff
- checked: selected_taxon_content_ids.include?(taxon.content_id),
+ checked: checked,
```

But now Whitehall is on Ruby 3.1, we can use [shorthand hash syntax](https://dev.to/baweaver/ruby-3-1-shorthand-hash-syntax-first-impressions-19op).

So `{ checked: checked }` becomes `{ checked: }`

And Rubocop actually enforces this new style – so we have to use it.

Comment (DilwoarH) [reply lib/taxonomy/topic_taxonomy.rb:48]: That's not very readable...

**Review by DilwoarH — APPROVED** [review]

**Pull request merged**